frontend/tests/utils/helpers.ts

```diff
@@ -17,19 +33,22 @@ export async function tryUntil(assert: () => void) {
 
   let retry = 0;
 
+  let savedError: Error = new Error();
+
   while (retry < MAX_RETRY) {
     try {
       await assert();
 
       break;
     } catch (error) {
       retry += 1;
-
+      savedError = error;
       await page.waitForTimeout(WAIT_FOR_MS);
     }
   }
 
   if (retry === MAX_RETRY) {
-    throw Error("tryUntil() assertion failed!");
+    savedError.message += " tryUntil() failed";
+    throw savedError;
```

Comment:
I was receiving the,
> tryUntil() failed

message and it wasn't quite useful,  instead threw the last saved error